src/views/common/Animation/AnimationControls/PlaybackBookmarkButton.tsx

```diff
@@ -0,0 +1,38 @@
+import { faBookmark as hollow } from '@fortawesome/free-regular-svg-icons'
+import { faBookmark as solid } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import React, { useCallback, useMemo } from 'react'
+
+
+export const BOOKMARK_BUTTON = "bookmarkButton"
+
+type PlaybackSyncButtonProps = {
+    handleSave: () => void
+    stateIsSaved: boolean
```

Comment:
This component needs only a (parameterless) callback to execute when the user wants to bookmark the current state, and a boolean value identifying whether the current state is a bookmark (in which case we show a different icon and ignore requests to bookmark again). There isn't a way to delete bookmarks under the current implementation, but it would be a minor addition.